File: C++(3)/3.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n;
    cout << "Ingrese el tamaño de la matriz identidad: ";
    cin >> n;

    vector<vector<int>> matriz(n, vector<int>(n, 0));

    // Generar matriz identidad
    for (int i = 0; i < n; i++) {
        matriz[i][i] = 1;
    }

    // Mostrar matriz
    cout << "Matriz identidad de tamaño " << n << ":" << endl;
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            cout << matriz[i][j] << " ";
        }
        cout << endl;
    }

    return 0;
}

File: Py (3)/3.py
def matriz_identidad():
    n = int(input("Ingrese el tamaño de la matriz identidad: "))

    matriz = [[0 for j in range(n)] for i in range(n)]

    # Generar matriz identidad
    for i in range(n):
        matriz[i][i] = 1

    ################
    print(f"Matriz identidad de tamaño {n}:")
    for fila in matriz:
        print(" ".join(str(x) for x in fila))

matriz_identidad()
